pkg/lib/authn/identity/loginid/provider.go

```diff
@@ -85,6 +85,38 @@ func (p *Provider) GetByValue(value string) ([]*identity.LoginID, error) {
 	return is, nil
 }
 
+func (p *Provider) GetByKeyAndValue(key string, value string) (*identity.LoginID, error) {
+	// Normalize expects loginID is in correct type so we have to validate it first.
+	invalid := p.Checker.ValidateOne(identity.LoginIDSpec{
+		Key:   key,
+		Type:  model.LoginIDKeyType(key),
+		Value: value,
+	}, CheckerOptions{
+		// Admin can create email login id which bypass domains blocklist allowlist
+		// it should not affect getting identity
+		// skip the checking when getting identity
+		EmailByPassBlocklistAllowlist: true,
+	})
+
+	if invalid != nil {
+		return nil, invalid
+	}
+
+	normalizer := p.NormalizerFactory.NormalizerWithLoginIDType(model.LoginIDKeyType(key))
+	normalizedloginID, err := normalizer.Normalize(value)
+	if err != nil {
+		return nil, err
```

Comment:
This is something I missed in the spec. If the value cannot be normalized, then we should return `ErrGetUsersInvalidArgument.New("invalid Login ID value")`. Please update the spec in this PR to reflect this corrected behavior.

Reply:
updated error handling in this function